src/components/CompletedSubReqCourse.vue

```diff
@@ -1 +1,184 @@
-<!-- Component for a completed SubReqCourse-->
+<template>
+  <div class="completedsubreqcourse">
+    <div
+      v-for="courseObject in courseObjects"
+      :key="courseObject.uniqueID"
+      class="completed-reqCourses-course-wrapper"
+    >
+      <div id="completedSeparator" class="separator"></div>
+      <div class="completed-reqCourses-course-heading-wrapper">
+        <div class="completed-reqCourses-course-heading-course">
+          <span class="completed-reqCourses-course-heading-check"
+            ><img src="@/assets/images/checkmark-green.svg"
+          /></span>
+          {{ courseLabel }}
+        </div>
+        <div class="completed-reqCourses-course-heading-reset-button reqCourse-button">
+          {{ resetText }}
+        </div>
+      </div>
+      <div class="completed-reqCourses-course-object-wrapper">
+        <course
+          v-bind="courseObject"
+          :courseObj="courseObject"
+          :id="courseObject.subject + courseObject.number"
+          :uniqueID="courseObject.uniqueID"
+          :compact="true"
+          :active="false"
+          :isCompletedReqCourse="true"
+          :isReqCourse="true"
+          class="completed-reqCourses-course-object"
+        />
+        <div class="completed-reqCourses-course-object-semester">
+          {{ semesterLabel }}
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue, { PropType } from 'vue';
+import firebase from 'firebase/app';
+import Course from '@/components/Course.vue';
+import { DisplayableRequirementFulfillment, CourseTaken } from '@/requirements/types';
+import {
+  AppCourse,
+  FirestoreSemesterCourse,
+  AppSemester,
+  FirestoreSemesterType,
+} from '@/user-data';
+
+Vue.component('course', Course);
+
+type CrseInfo = {
+  roster: string;
+  crseIds: number[];
+};
+
+type CompletedSubReqCourseSlot = {
+  isCompleted: true;
+  courses: readonly CourseTaken[];
+};
+
+type IncompleteSubReqCourseSlot = {
+  isCompleted: false;
+  courses: CrseInfo[];
+};
+
+type SubReqCourseSlot = CompletedSubReqCourseSlot | IncompleteSubReqCourseSlot;
+
+type Data = {
+  courseObjects: AppCourse[];
+  semesterType: FirestoreSemesterType;
+  semesterYear: number;
+};
+
+export default Vue.extend({
+  props: {
+    subReq: Object as PropType<DisplayableRequirementFulfillment>,
+    subReqCourseId: Number,
+    crsesTaken: Array as PropType<readonly CourseTaken[]>,
+    semesters: Array as PropType<readonly AppSemester[]>,
+  },
+  data(): Data {
+    return {
+      semesterType: 'Fall',
+      semesterYear: 1234,
```

Comment:
If this is default values that will never be seen by the user, can we set semesterType as '' and semesterYear as 0 instead? Think that would make it less confusing to understand

Reply:
@hanywang2 That's a good point. I made `semesterType` to be of type `FirestoreSemesterType` so it can only be `"Fall" | "Winter" | "Spring" | "Summer"`. I can change `semesterYear` to be 0 and I'll leave a comment for why `semesterType` is `'Fall'`.